src/libostree/ostree-repo.c

```diff
@@ -3113,6 +3113,27 @@ reload_remote_config (OstreeRepo          *self,
   return TRUE;
 }
 
+static gboolean
+reload_sysroot_config (OstreeRepo          *self,
+                       GCancellable        *cancellable,
+                       GError             **error)
+{
+  { g_autofree char *bootloader = NULL;
+
+    if (!ot_keyfile_get_value_with_default_group_optional (self->config, "sysroot",
+                                                           "bootloader", "auto",
+                                                           &bootloader, error))
+      return FALSE;
+
+    if (bootloader == NULL)
+      return glnx_throw (error, "Invalid empty bootloader configuration");
```

Comment:
I don't think this case can happen...did you mean to compare vs the empty string `""`?  Either way it should probably be instead something like:

```
if (!(g_str_equal (bootloader, "auto") || g_str_equal (bootloader, "none"))
  return glnx_throw (error, "Invalid bootloader configuration: '%s'", bootloader);
```

Reply:
Hmm, for some reason thought it would occur with `bootloader=` in the keyfile, but looks like this just returns an empty string.

Initially I was thinking that `reload_sysroot_config()` should allow any value for `bootloader=$value`, and then have `ostree_sysroot_write_deployments_with_options()` search for something that matches the value (e.g. if we later supported a `/etc/ostree/bootloaders`, `ostree_sysroot_write_deployments_with_options()` would search for `/etc/ostree/bootloaders/my_bootloader` for `bootloader=my_bootloader`). I agree now that validating right at the parsing for this change is better - can always be reworked if `/etc/ostree/bootloaders` or `/usr/lib/ostree/bootloaders` is implemented. Fixing!